### PR TITLE
- create step in Dockerfile and Dockerfile.tests, to copy smart contr…

### PR DESCRIPTION
**Pull request by tabakopavel**
- create step in Dockerfile and Dockerfile.tests, to copy smart contact build artifact to docker image for node

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,6 +5,12 @@ ARG PROFILE=release
 WORKDIR /cerenetwork
 COPY . /cerenetwork
 
+# Copy smart-contract artifacts (cere01.wasm and metadata.json)
+ARG SMART_CONTRACT_VERSION=1.1.0
+FROM cere-enterprise-smart-contracts/smart-contracts:$SMART_CONTRACT_VERSION as smart-contracts
```
Comment (alexremn) [inline Dockerfile]: Are you using multistage build or you need this image just to copy files?
If the second one, so --from could be specified with image name.
Also, image name should be specified fully, with ECR repo.

File: Dockerfile.tests [hunk 2]
```diff
@@ -4,6 +4,12 @@ LABEL description="This is the build stage to create the binary."
 WORKDIR /cerenetwork
 COPY . /cerenetwork
 
+# Copy smart-contract artifacts (cere01.wasm and metadata.json)
+ARG SMART_CONTRACT_VERSION=1.1.0
+FROM cere-enterprise-smart-contracts/smart-contracts:$SMART_CONTRACT_VERSION as smart-contracts
```
Comment (alexremn) [inline Dockerfile.tests]: The same as above.

**Review by alexremn — CHANGES_REQUESTED** [review]

**Review by alexremn — APPROVED** [review]

File: Dockerfile.tests [hunk 3]
```diff
@@ -4,6 +4,12 @@ LABEL description="This is the build stage to create the binary."
 WORKDIR /cerenetwork
 COPY . /cerenetwork
 
+# Copy smart-contract artifacts (cere01.wasm and metadata.json)
+ARG SMART_CONTRACT_VERSION=1.1.0
+ARG ECR_REGISTRY
+COPY --from=$ECR_REGISTRY/crb-smart-contracts:$SMART_CONTRACT_VERSION /smart-contracts/cere01/target/ink/cere01.wasm /cerenetwork/frame/ddc-metrics-offchain-worker/src/tests/test_data/
```
Comment (naure) [inline Dockerfile.tests]: It is `cere02` that is needed here.

**Comment by naure** [comment]
Great!

Can you also do these:
* Review the file `tests/mod.rs` and verify (or update) the filenames (search for `test_data/ddc.wasm` and `test_data/metadata.json`).
* Remove the existing files from git (ddc.wasm and metadata.json).
* Add the paths to .gitignore
* Add an instruction in the README for developers with an example command to copy the files (maybe `docker cp …` equivalent to `COPY`)

File: Dockerfile.tests (hunk 3 above)
Comment (tabakopavel) [reply Dockerfile.tests]: But in README for SC : take cere01 or cere02, doesn't matter 
IS it correct ?

Because i took  cere01. Do we need cere02 ?

Comment (naure) [reply Dockerfile.tests]: Yes use cere02 everywhere.

Comment (naure) [reply Dockerfile.tests]: I know, it’s a terrible name.

**Pull request merged**